src/commands_refactor.py

```diff
@@ -0,0 +1,241 @@
+import random
+
+import psycopg2
+import psycopg2.extras
+from psycopg2 import sql
+from src.modules.catfact_helper import get_catfact
+from src.modules.db_helper import member_exists, insert_member, get_table, connection_error, dbfunc_run
+from src.modules.discord_helper import change_nickname, kick_member, try_send
+from src.tools.botfunction import BotFunction
+from src.tools.message_return import message_data
+
+
+class command_on_message(BotFunction):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    async def action(self, message):
+        raise NotImplementedError
+
+
+class catfact(command_on_message):
+    """
+    $catfact
+    Gets random catfact
+    """
+    def __init__(self,*args,**kwargs):
+        super().__init__(*args,**kwargs)
+
+    async def action(self, message, *args, **kwargs):
+        return message_data(message.channel, get_catfact())
+  
+class debug(command_on_message):
+    """
+    $debug
+    activates debug mode 
+    """
+    def __init__(self,*args,**kwargs):
+        self.roleslist = ["Admins", "Yangbot Devs", "Server Legacy"]
+        super().__init__(roles = self.roleslist, role_whitelist = True, *args, **kwargs)
+    
+    async def action(self, message, *args, **kwargs):
+        if self.bot.debug is False:
```

Comment:
Make sure all the functions memory (like the repeated messages auto_on_message) is deleted when debug action is taken